src/library_pthread.js

```diff
@@ -510,19 +510,11 @@ var LibraryPThread = {
     }
   },
 
-#if USE_PTHREADS
   _num_logical_cores__deps: ['emscripten_force_num_logical_cores'],
   _num_logical_cores: '; if (ENVIRONMENT_IS_PTHREAD) __num_logical_cores = PthreadWorkerInit.__num_logical_cores; else { PthreadWorkerInit.__num_logical_cores = __num_logical_cores = allocate(1, "i32*", ALLOC_STATIC); HEAPU32[__num_logical_cores>>2] = navigator["hardwareConcurrency"] || ' + {{{ PTHREAD_HINT_NUM_CORES }}} + '; }',
-#else
-  _num_logical_cores: '{{{ makeStaticAlloc(1) }}}',
-#endif
 
   emscripten_has_threading_support: function() {
-#if USE_PTHREADS
-    return typeof SharedArrayBuffer !== 'undefined';
-#else
-    return 0;
-#endif
+    return true;
```

Comment:
I general I'm not a fan of a code that "it not used today but might become relevant again in the future".  Not only is it dead code, but it can mislead the reader. 
I guess I don't mind reverting this, with perhaps a clear comment that it will always be true and why?

Reply:
Yeah, at minimum the code will need to read

```javascript
emscripten_has_threading_support: function() {
#if USE_PTHREADS
    return true;
#else
    return false;
#endif
}
```

and in the fringe case where one is a) running with a browser that does not support SharedArrayBuffer, b) the page fails to load (or startup is set to manual), and c) user JS code calls `_emscripten_has_threading_support()` function, it would erroneously return `true` - there is nothing wrong with `return typeof SharedArrayBuffer !== 'undefined';` here, so good to keep that way. I think `emscripten_has_threading_support()` is best to return true if current browser supports SAB and current page was built with multithreading active.

Reply:
I think that by definition this file can only be included with USE_PTHREADS.. otherwise the _stub file gets included and this file does not.  So to have this `ifdef` at all is a little misleading to the reader.. unless I'm missing something?

So how about we just make it:

```
emscripten_has_threading_support: function() {
  return typeof SharedArrayBuffer !== 'undefined';
}
```

Reply:
Oh yeah, great point - I was going by static reading of the function, forgot about `library_pthread_stub.js` altogether. Your proposed form looks good.